unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,176 @@
+    def register_fixture(self):
+        initial = RedisClientBase.register
+        RedisClientBase.register = MagicMock()
+        test = RedisClientBase(area_id=area_id, autoregister=False)
+        yield test
+        RedisClientBase.register = initial
+
+    @pytest.fixture()
+    def unregister_fixture(self):
+        initial = RedisClientBase.unregister
+        RedisClientBase.unregister = MagicMock()
+        test = RedisClientBase(area_id=area_id, autoregister=False)
+        yield test
+        RedisClientBase.unregister = initial
+
+    @pytest.fixture()
+    def on_register_fixture(self):
+        initial = RedisClientBase._on_register
+        RedisClientBase._on_register = MagicMock()
+        test = RedisClientBase(area_id=area_id, autoregister=False)
+        yield test
+        RedisClientBase._on_register = initial
+
+    @pytest.fixture()
+    def check_buffer_message_fixture(self):
+        initial = RedisClientBase._check_buffer_message_matching_command_and_id
+        RedisClientBase._check_buffer_message_matching_command_and_id = MagicMock()
+        test = RedisClientBase(area_id=area_id, autoregister=False)
+        yield test
+        RedisClientBase._check_buffer_message_matching_command_and_id = initial
+
+    @pytest.fixture()
+    def check_transaction_id_cached_out_call_success_fixture(self):
+        initial = RedisClientBase._check_transaction_id_cached_out
+        RedisClientBase._check_transaction_id_cached_out = MagicMock()
+        test = RedisClientBase(area_id=area_id, autoregister=False)
+        yield test
+        RedisClientBase._check_transaction_id_cached_out = initial
+
+    def test_function_check_buffer_message_matching_command_and_id_fixture_call_success(self,
+                                                                                        check_buffer_message_fixture):
+        check_buffer_message_fixture._check_buffer_message_matching_command_and_id()
+        check_buffer_message_fixture._check_buffer_message_matching_command_and_id.assert_called_once()
+
+    def test_function_register_call_success(self, register_fixture):
+        register_fixture.register()
+        register_fixture.register.assert_called_once()
+
+    def test_function_unregister_call_success(self, unregister_fixture):
+        unregister_fixture.unregister()
+        unregister_fixture.unregister.assert_called_once()
+
+    def test_function_on_register_call_success(self, on_register_fixture):
+        on_register_fixture._on_register()
+        on_register_fixture._on_register.assert_called_once()
+
+    def test_function__check_transaction_id_cached_out_call_success(self,
+                                                                    check_transaction_id_cached_out_call_success_fixture):
+        check_transaction_id_cached_out_call_success_fixture._check_transaction_id_cached_out()
+        check_transaction_id_cached_out_call_success_fixture._check_transaction_id_cached_out.assert_called_once()
```

Comment:
This comment: https://github.com/gridsingularity/d3a-api-client/pull/163/files#r691428221 applies to all of these.